crypto/evp/m_sigver.c

```diff
@@ -176,6 +176,13 @@ static int do_sigver_init(EVP_MD_CTX *ctx, EVP_PKEY_CTX **pctx,
         }
 
         if (mdname != NULL) {
+            /*
+             * If the old digest was a legacy one then we should ensure we free
+             * any md_data. This can be removed once no more legacy.
+             */
+            if (ctx->digest != NULL && ctx->digest->ctx_size > 0)
+                OPENSSL_clear_free(ctx->md_data, ctx->digest->ctx_size);
+            ctx->md_data = NULL;
```

Comment:
Why not `EVP_MD_CTX_reset`?

Reply:
`EVP_MD_CTX_reset` would reset *everything* which is too much in this context (for example we already set up the pctx). However your question made me double check things, and I found that we probably also need to call the cleanup() function. Fixup pushed.

Reply:
Would you mind making that an internal function that's used both by `EVP_MD_CTX_reset` and this spot?

(why code copying is so popular, I will never understand)

Reply:
> Would you mind making that an internal function that's used both by EVP_MD_CTX_reset and this spot?

Ok done.